packages/contracts/src/make-dump.ts

```diff
@@ -0,0 +1,93 @@
+/* External Imports */
+import { Signer } from 'ethers'
+import {
+  computeStorageSlots,
+  getStorageLayout,
+} from '@defi-wonderland/smock/dist/src/utils'
+
+/* Internal Imports */
+import { predeploys } from './predeploys'
+import { getContractArtifact } from './contract-artifacts'
+
+export interface RollupDeployConfig {
+  whitelistConfig: {
+    owner: string | Signer
+    allowArbitraryContractDeployment: boolean
+  }
+  gasPriceOracleConfig: {
+    owner: string | Signer
+    initialGasPrice: number
+  }
+  l1StandardBridgeAddress: string
+  l1FeeWalletAddress: string
+  l1CrossDomainMessengerAddress: string
+}
+
+/**
+ * Generates the initial state for the L2 system by injecting the relevant L2 system contracts.
+ *
+ * @param cfg Configuration for the L2 system.
+ * @returns Generated L2 genesis state.
+ */
+export const makeStateDump = async (cfg: RollupDeployConfig): Promise<any> => {
+  const variables = {
```

Comment:
Perhaps `RollupDeployConfig` should be merged with this `variables` object?

Reply:
I wanted to create an abstraction here so we don't have to change the config whenever we change internal variables in the smart contracts